### DISCUSSION TITLE
what scope do I need to get groups in Google SSO?

### DISCUSSION THREAD

**Discussion by BrianHicks** [argoproj/argo-workflows#7665] [Q&A]
Similar to https://github.com/argoproj/argo-workflows/discussions/5303, I'm trying to set up SSO for group-based RBAC using Google.

I've added the scope that Argo CD uses (`https://www.googleapis.com/auth/admin.directory.group.readonly`) and no group information shows up in my user view. Trying the more naive `groups` value as shown in the sample YAML results in a failure as well (Google rejects it.)

So, any ideas? What scope do I need to add? Do I need to add more flags to the server?

I'm running Argo Workflows v3.2.6 with the `--auth-mode sso` flag on the server already. I can log in with SSO, so that much at least is working!

**Comment by haghabozorgi** [comment]
for anyone else who finds themselves here, 
google does not support group claim. I believe these are the list of supported claims available. https://accounts.google.com/.well-known/openid-configuration. 
If you note how dex does google auth group/rbac support (e.g argocd), you will notice it requires a separate service account to be able to query the admin api for group membership. I believe to use google you will likely need to rely on dex (perhaps if you already have argocd running), or hack something together? Not sure if argo team plans on supporting the dex setup for workflows directly in the future?
